synthetic_data/synthetic_data.py

```diff
@@ -305,9 +306,112 @@ def make_tabular_data(
     #
 
     if noise_level_x > 0.0:
-        x_noise = generate_x_noise(
-            x_final[:, :n_informative], noise_level_x, seed=seed
-        )
+        x_noise = generate_x_noise(x_final[:, :n_informative], noise_level_x, seed=seed)
         x_final[:, :n_informative] = x_final[:, :n_informative] + x_noise
 
     return x_final, y_reg, y_prob, y_labels
+
+
+def make_data_from_report(
+    report: dict,
+    n_samples: int = None,
+    noise_level: float = 0.0,
+    seed=None,
+) -> pd.DataFrame:
+    """
+    Use a DataProfiler report to generate a synthetic data set to mimic the report.
+    args:
+        report (dict) - DataProfiler report
+        n_samples (int) - number of samples to generate
+        noise_level (float) - level of white noise (jitter) added to x
+        seed - numpy random state object for repeatability
+
+    returns X: DataFrame of shape [n_samples, n_total]
+    """
+
+    # make sure correlation matrix was generated
+    if report["global_stats"]["correlation_matrix"] is None:
+        raise Exception("The report must have the correlation matrix enabled")
+
+    # make sure no non-numerical columns exist
+    for stat in report["data_stats"]:
+        if stat["data_type"] not in ["int", "float"]:
+            raise Exception("The function only supports numerical variables")
+
+    # if n_samples not provided, generate same samples as original dataset
+    if not n_samples:
+        n_samples = report["global_stats"]["samples_used"]
+
+    n_informative = len(report["data_stats"])
+    # setting to 0 to skip logic below
+    n_redundant = 0
```

Comment:
@JGSweets @tazitoo hard code this or pass the parameter? Would someone ever want a profile + noise + nuisance?